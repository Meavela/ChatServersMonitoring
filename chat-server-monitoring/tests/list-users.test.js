import React from 'react';
import { render } from '@testing-library/react';
import ListUsers from '../src/Components/list-users';
import renderer from 'react-test-renderer';

// Mock the component to simulate the API
jest.mock('../src/Components/list-users');

// Check that the Users page display the right things
test('Display List User page', () => {
    const {container} = render(<ListUsers />)
    expect(container.textContent).toMatch('UsersServerUser IDUsernameStatus151.11.52.681LouACTIVE172.82.64.792Marc-AntoineACTIVE')
});

// Check that the Users page don't change
test('Not change List User page', () => {
    const component = renderer.create(
        <ListUsers />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();  
});