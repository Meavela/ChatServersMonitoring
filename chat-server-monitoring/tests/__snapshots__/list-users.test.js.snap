// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not change List User page 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root list-card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root title"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Users
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root list-card-content"
    >
      <div
        className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          aria-label="simple table"
          className="MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Server
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                User ID
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Username
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                151.11.52.68
              </th>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                1
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                Lou
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                ACTIVE
              </td>
            </tr>
            <tr
              className="MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                172.82.64.79
              </th>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                Marc-Antoine
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                ACTIVE
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
